overlord/ifacestate/helpers.go

```diff
@@ -114,22 +110,12 @@ func (m *InterfaceManager) addSnaps() error {
 }
 
 func (m *InterfaceManager) profilesNeedRegeneration() bool {
```

Comment:
this is run too early if we are seeded,  and it's a problem even on classic, we probably need to return false if we are not seeded, and trigger a variant of writing the key when we are about to setup profiles for core for the first time?